services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java

```diff
@@ -690,14 +690,17 @@ private void doDeleteMeter(CommandMessage message, String replyToTopic, Destinat
 
         try {
             DatapathId dpid = DatapathId.of(request.getSwitchId().toLong());
-            long txId = context.getSwitchManager().deleteMeter(dpid, request.getMeterId());
+            context.getSwitchManager().deleteMeter(dpid, request.getMeterId());
 
-            DeleteMeterResponse response = new DeleteMeterResponse(txId != 0L);
+            boolean deleted = context.getSwitchManager().dumpMeters(dpid)
+                    .stream()
+                    .anyMatch(config -> config.getMeterId() == request.getMeterId());
+            DeleteMeterResponse response = new DeleteMeterResponse(deleted);
             InfoMessage infoMessage = new InfoMessage(response, System.currentTimeMillis(), message.getCorrelationId(),
                     replyDestination);
             producerService.sendMessageAndTrack(replyToTopic, infoMessage);
         } catch (SwitchOperationException e) {
-            logger.info("Meter deletion is unsuccessful. Switch {} not found", request.getSwitchId());
+            logger.info(String.format("Failed to delete the meter from switch %s", request.getSwitchId()), e);
```

Comment:
I don't think we need traceback here. `SwitchOperationException` is our exception and we know when and why it happens.

PS Info level log record with traceback is weird (as for me).